server/events/vcs/github_pull_status_fetcher.go

```diff
@@ -0,0 +1,51 @@
+package vcs
+
+import (
+	"github.com/google/go-github/v31/github"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/models"
+)
+
+type PullReqStatusFetcher interface {
+	FetchPullStatus(repo models.Repo, pull models.PullRequest) (models.PullReqStatus, error)
+}
+
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_github_pull_status_fetcher.go PullApprovalChecker
+type PullApprovalChecker interface {
+	GetRepoStatuses(repo models.Repo, pull models.PullRequest) ([]*github.RepoStatus, error)
+	PullIsApproved(repo models.Repo, pull models.PullRequest) (bool, error)
+	PullIsSQMergeable(repo models.Repo, pull models.PullRequest, statuses []*github.RepoStatus) (bool, error)
+	PullIsLocked(baseRepo models.Repo, pull models.PullRequest, statuses []*github.RepoStatus) (bool, error)
```

Comment:
Why didn't you just add these methods to `IGithubClient` and then just use that in the `SQBasedPullStatusFetcher`?

Reply:
Hmm I was thinking that doing so would make testing easier but since we're already using `IGithubClient` for testing, it makes sense to just add these methods to `IGithubClient` and use the same interface in `SQBasedPullStatusFetcher`.